### PR TITLE
Refactor plot styling

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-543

#### Link to staging deployment URL 

Coming soon

#### Links to any Pull Requests related to this

N.A.

#### Anything else the reviewers should know about the changes here

- IN PROGRESS
- Pused to PR to request for comments on architecture / concerns
- Right now only implemented for Continuous Embedding plot. Once component is ok, other plots will be updated.

# Changes
### Code changes

- `PlotStyling` is refactored out of `plots-and-tables` folder into `components`, under a new `plotStyling` folder.
- All components of plot styling also moved to the `plotStyling` folder
- `PlotStyling` component interface :  
-- `formConfig` : receive configuration object to control form fields that should be used
-- `config` : receive plot config
-- `onUpdate` : function to update `config` object

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [X] Tested locally with Inframock
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/components/plotStyling/PlotStyling.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  const ComponentMapping = {
+    dimensions: <DimensionsRangeEditor config={config} onUpdate={onUpdate} />,
+    title: <TitleDesign config={config} onUpdate={onUpdate} />,
+    font: <FontDesign config={config} onUpdate={onUpdate} />,
+    axes: <AxesDesign config={config} onUpdate={onUpdate} />,
+    colourbar: <ColourbarDesign config={config} onUpdate={onUpdate} />,
+    colourInversion: <ColourInversion config={config} onUpdate={onUpdate} />,
+    marker: <PointDesign config={config} onUpdate={onUpdate} />,
+    legend: <LegendEditor onUpdate={onUpdate} config={config} />,
+  };
```
Comment (aerlaut) [inline src/components/plotStyling/PlotStyling.jsx]: Mappings between component and key to use in a plot config object

File: src/components/plotStyling/PlotStyling.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            {
+              Object.getOwnPropertyDescriptor(el, 'children')
+                && el.children.length > 0
+                ? buildForm(el.children)
+                : ''
+            }
+          </Panel>
+        </Collapse>
+      );
+    }
```
Comment (aerlaut) [inline src/components/plotStyling/PlotStyling.jsx]: Function to build form object

File: src/pages/experiments/[experimentId]/plots-and-tables/embedding-continuous/index.jsx [hunk 3]
```diff
@@ -118,6 +110,39 @@ const EmbeddingContinuousPlot = () => {
...
+    {
+      panel: 'Marker',
+      form: ['marker'],
+    },
+    {
+      panel: 'Legend',
+      form: ['legend'],
+    },
+  ];
+
```
Comment (aerlaut) [inline src/pages/experiments/[experimentId]/plots-and-tables/embedding-continuous/index.jsx]: Example of plot config

Comment (aerlaut) [reply src/pages/experiments/[experimentId]/plots-and-tables/embedding-continuous/index.jsx]: I think I'll change `panel` to `panelTitle`, and `form` to `stylingOption`

File: src/components/plotStyling/PlotStyling.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    <>
+      {
+        buildForm(formConfig)
+      }
+    </>
+  );
+};
+
+PlotStyling.propTypes = {
+  components: PropTypes.array,
```
Comment (ivababukova) [inline src/components/plotStyling/PlotStyling.jsx]: this prop is unused, you can remove it

Comment (aerlaut) [reply src/components/plotStyling/PlotStyling.jsx]: That's supposed to be renamed to `formConfig`. I'll fix it in next commit

**Comment by aerlaut** [comment]
Changed to another PR based on more recent `master` branch

**Pull request closed without merging**